### PR TITLE
feat(calling): pinch to zoom in/out a video stream (AR-3231)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-3231" title="AR-3231" target="_blank">StoryAR-3231</a>  Zoom in/out on maximised video stream
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Description

With this PR, users in a call can now pinch to zoom in/out a video stream

https://github.com/wireapp/wire-android-reloaded/assets/18124919/c510e2ad-007a-4829-afa6-7ff213ca99f3

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1770/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1770/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1770/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6483781)(6483781)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6483781)(6483781)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1770/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1770/5/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/participantsview/ParticipantsTiles.kt [hunk 1]
```diff
@@ -153,7 +153,7 @@ private fun pagesCount(size: Int): Int {
 }
 
 private const val MAX_TILES_PER_PAGE = 8
-private const val MAX_ITEMS_FOR_ONE_ON_ONE_VIEW = 2
```
Comment (tmspzz) [inline app/src/main/kotlin/com/wire/android/ui/calling/ongoing/participantsview/ParticipantsTiles.kt:156]: Is this right?

**Review by tmspzz — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1770/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6517382)(6517382)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6517382)(6517382)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/fullscreen/FullScreenTile.kt [hunk 2]
```diff
@@ -43,27 +46,29 @@ fun FullScreenTile(
...
-                    start = dimensions().spacing4x,
-                    end = dimensions().spacing4x
-                ),
-            participantTitleState = it,
-            isSelfUser = selectedParticipant.isSelfUser,
-            shouldFill = false,
-            onSelfUserVideoPreviewCreated = sharedCallingViewModel::setVideoPreview,
-            onClearSelfUserVideoPreview = sharedCallingViewModel::clearVideoPreview
-        )
+        Box(modifier = Modifier) {
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/calling/ongoing/fullscreen/FullScreenTile.kt]: Unless we are really adding something to the modifier I don't think we need to have a "base"/"default" one here

**Review by alexandreferris — COMMENTED** [review]
Nice work! Just a tiny UI comment on the Box Modifier and +1 on Tommaso's comment

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/participantsview/ParticipantsTiles.kt (hunk 1 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/calling/ongoing/participantsview/ParticipantsTiles.kt:156]: yes 3 is the correct value.. the value got changed in CBR indicator PR

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/fullscreen/FullScreenTile.kt (hunk 2 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/calling/ongoing/fullscreen/FullScreenTile.kt]: yes leftover, needs cleanup

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1770/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6517620)(6517620)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6517620)(6517620)-staging-compat.apk)

**Pull request merged**